### ISSUE TITLE
ASP.NET Core client side Blazor applications can't be run in VSCode

### ISSUE THREAD

**Issue by NTaylorMullen** [OmniSharp/omnisharp-vscode#3024]
This was originally reported over in https://github.com/aspnet/AspNetCore/issues/9451 and it seems that the dialog to add build assets to the project can't detect Blazor client side applications because they're not true runnable applications on their own.

To fix this there's two steps:
1. Make runnable project detection understand Blazor client side applications
2. Update the run .net core console app to utilize `dotnet run --no-build` instead of `dotnet *.dll`. The distinction about using `dotnet run` enables the Blazor project to hijack the run process and properly boot a runnable application.

/cc @danroth27

**Comment by gregg-miskelly** [comment]
I don't think this plan will work -- launch.json configurations need to be for both running and debugging and, at least the .NET Debugger, will not be able to work if you launch the problem using `dotnet run`.

**Comment by NTaylorMullen** [comment]
Hmmm interesting. Why is it that `dotnet run` can't be debugged?

**Comment by gregg-miskelly** [comment]
`dotnet run` will launch the target in a child process. The .NET Debugging APIs have no support for child process debugging.

**Comment by gregg-miskelly** [comment]
Actually, thinking more - this is a much bigger problem than just tweaking launch.json a little. 

For a pure client app - what you want F5 to do is start the Blazor web server not under a debugger, and then after it is ready, launch Chrome (or whatever supported web browser), and connect the Chrome debug adapter to it.

For a 'Blazor (ASP.NET Core hosted)' app it might make sense to debug both at the same time.

**Comment by NTaylorMullen** [comment]
I see I see, ya good point. @SteveSandersonMS / @danroth27 is Blazor debugging in the browser in a viable state?

**Comment by danroth27** [comment]
@NTaylorMullen Debugging Blazor in the browser is still pretty rough. Eventually thought we do want to have an integrated experience where you can just F5 and have VS launch the browser with remote debugging enabled and then correctly attach.

**Comment by danroth27** [comment]
For comparison, in VS we also don't support F5 for client-side Blazor apps. You have to Ctrl-F5, and it's been that way for over a year now.

**Comment by gregg-miskelly** [comment]
Here are my suggestions:
1. Don't do anything right now in terms of launch.json generation. We can leave this bug open to consider doing something someday for it.
2. Update the documentation to point point at running the `dotnet watch` task.
